dali/pipeline/operator/builtin/external_source.h

```diff
@@ -148,7 +144,7 @@ class ExternalSource : public Operator<Backend> {
       copy_to_storage_event = copy_to_storage_events_.GetEmpty();
     }
 
-    data.front()->Copy(tl, stream);
+    data.front()->Copy(t, stream);
     if (std::is_same<SrcBackend, GPUBackend>::value) {
       cudaEventRecord(*copy_to_storage_event.front(), stream);
```

Comment:
hmm, shouldn't we wait for any event other than CPU->CPU (sync copy)?

Reply:
Added waiting for GPU to anything, CPU pinned to GPU.

Reply:
Cool. Thanks for adding the comment.